Automattic/Gravatar-SDK-Android
# QuickEditor: Upload image from the gallery

Closes #226

### Description

This PR adds an option to pick an image from the phone gallery and upload it to the server after cropping. 

For the image picking I went with the latest recommended solution from Google - [Photo Picker](https://developer.android.com/training/data-storage/shared/photopicker).
For the cropping, I used the UCrop library as we did already for the WordPress picker. 

I've mimicked the web by disabling the `Upload image` button while the avatar is being uploaded to the web.


https://github.com/user-attachments/assets/9b5df0cd-2504-4929-a79c-e46da2e081c7



### Testing Steps
**Important Note**: The backend is currently returning the wrong format for the `updated date` field. As a workaround, you can apply this patch to comment that field from the Avatar object.
[Gravatar-Android-13-04-48.patch](https://github.com/user-attachments/files/16455284/Gravatar-Android-13-04-48.patch)

1. Make sure you have at least two avatars uploaded on Gravatar
2. Run the DemoApp 
3. Go to Avatar Update tab
4. Tap `Update Avatar` button
5. Follow the steps to log in (if necessary)
8. Tap the `Upload Image` button
9. Tap the `Choose a Photo`
10. Pick a photo from the gallery
11. Crop if you want and confirm
12. Confirm you see the photo in the list with the progress indicator
13. Confirm the photo was uploaded and can be selected
